packager/mpd/base/period.cc

```diff
@@ -240,11 +239,12 @@ const AdaptationSet* Period::FindOriginalAdaptationSetForTrickPlay(
   MediaInfo media_info_no_trickplay = media_info;
   media_info_no_trickplay.mutable_video_info()->clear_playback_rate();
 
-  std::string key = GetAdaptationSetKey(media_info_no_trickplay);
+  std::string key = GetAdaptationSetKey(media_info_no_trickplay, false);
   const std::list<AdaptationSet*>& adaptation_sets =
       adaptation_set_list_map_[key];
   for (AdaptationSet* adaptation_set : adaptation_sets) {
-    if (protected_adaptation_set_map_.Match(*adaptation_set, media_info)) {
+    if (protected_adaptation_set_map_.Match(*adaptation_set, media_info,
+                                            true)) {
```

Comment:
Looks like this is a bug in the original code. The flag content_protection_in_adaptation_set should have the same value as the one in line 93.

Can you pass content_protection_in_adaptation_set to SetNewAdaptationSetAttributes then to  FindOriginalAdaptationSetForTrickPlay?

Reply:
Yes.